ImageD11/parameters.py

```diff
@@ -278,12 +278,11 @@ def saveparameters(self, filename):
         """
         Write parameters to a file
         """
-        f = open(filename, "w")
-        keys = list(self.parameters.keys())
-        keys.sort()
-        for key in keys:
-            f.write("%s %s\n" % (key, str(self.parameters[key])))
-        f.close()
+        with open(filename, "w") as f:
```

Comment:
This is the recommended syntax since Python 2.5 to avoid explicit call to `close` and ensure file closing on error: https://docs.python.org/2.7/library/stdtypes.html#file.close